configure.ac

```diff
@@ -347,6 +347,18 @@ if test "${have_openssl}" = "yes"; then
 	LIBS="${old_LIBS}"
 fi
 
+PKG_CHECK_MODULES([OPENSSL], [openssl >= 1.1.1], [have_openssl_111="yes"], [have_openssl_111="no"])
+if test "${have_openssl_111}" = "yes" ; then
+	old_LIBS="${LIBS}"
+	LIBS="${LIBS} ${OPENSSL_LIBS}"
+	AC_CHECK_FUNC(
+		[ENGINE_new],
+		[use_openssl_pkey_methods="yes"],
+		[use_openssl_pkey_methods="no"]
+	)
```

Comment:
Unfortunately use of pkey methods has necessitated some extra glue code between OpenSSL and PKCS#11. More on that below.
 
I think OpenSSL can be built with no engine support in which case it will define OPENSSL_NO_ENGINE. engine.h is always available so we can include it without checking, but is just a shell in that case.

Will take care of all other code/style comments and push fixups.

For other comments/questions, see below.